src/helpers/domainFindOptionsHelper.ts

```diff
@@ -0,0 +1,136 @@
+import { HttpRequest } from "@azure/functions";
+import { SortDirection } from "mongodb";
+import { DomainFindOptions } from "@zero-tech/data-store-core";
+import { DomainSortDirection, domainReflectionSchema } from "../types";
+declare type AscDesc = -1 | 1;
+declare type IncludeExlucde = 0 | 1;
+interface dynamicObject {
+  [key: string]: any;
+}
+const defaultProjection = ["domainid", "owner", "name"];
+const defaultSort = ["label", "owner"];
+const defaultSortDirection: AscDesc[] = [-1, -1];
+
+/*
+ * Generates a DomainFindOptions object based on query string parameters
+ * @param req - An HttpRequest
+ * @returns  - a DomainFindOptions object, with either default values or values supplied from the query string
+ */
+export function getDomainFindOptionsFromQuery(req: HttpRequest) {
+  //Get variables to project, matching the case insensitive names to case sensitive object names
+  const projectionObject = createProjection(req);
+  const sortObject = createSort(req);
+
+  let findOptions: DomainFindOptions = {
+    sort: sortObject,
+    projection: projectionObject,
+    limit: isNaN(+req.query.limit) ? 100 : +req.query.limit,
```

Comment:
Let's avoid setting a default limit value.  The only way to retrieve all records in a mongo query is to omit this field. By defaulting the limit we remove that option from the consumer.

Reply:
I think we really need to set a limit value or our db is gonna get killed, is there a valid use case that someone needs 20,000 domains in a single request

Reply:
Spoke outside the PR:
We decided that allowing users to supply a limit of 0, will allow them to get all the records from mongo.